### PR TITLE
Script to run local e2e test

### PR DESCRIPTION
**Pull request by bowei**
hack/run-local-e2e.sh will setup your environment and
write some scripts to help run e2e tests from a locally built
binary.

Usage

$ ./hack/run-local-e2e.sh my-cluster
...
$ bash /tmp/run-glbc.sh # run proxy and controller

### CODE REVIEW

**Comment by bowei** [comment]
@rramkumar1 @spencerhance

File: hack/run-local-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    net=$(echo ${net} | sed 's+.*networks/\([-a-z]*\).*$+\1+')
+    subnet=$(echo ${subnet} | sed 's+.*subnetworks/\([-a-z]*\)$+\1+')
+    project=$(echo ${selfLink} | sed 's+.*/projects/\([-a-z]*\)/.*+\1+')
+}
+
+parseInstance() {
+    local name=$1
+    local zone=$2
+    # Globals.
+    nodeTag=$(gcloud compute instances describe gke-test-default-pool-2b81ea3d-sw3l --zone us-west2-b --format='value(tags.items[0])')
```
Comment (rramkumar1) [inline hack/run-local-e2e.sh:23]: fix this? Seems to be hardcoded.

File: hack/run-local-e2e.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
+#/bin/bash
```
Comment (rramkumar1) [inline hack/run-local-e2e.sh:1]: should this be called run-local.sh? It's just running the controller, not the e2e tests.

File: hack/run-local-e2e.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+clusterName="$1"
+if [ -z ${clusterName} ]; then
+    echo "Usage: $0 CLUSTER_NAME"
+    exit 1
+fi
+
+fmt='value(networkConfig.network,networkConfig.subnetwork,zone,selfLink,name)'
+parseCluster $(gcloud container clusters list --format=${fmt} | grep "${clusterName}\$")
```
Comment (rramkumar1) [inline hack/run-local-e2e.sh:33]: if there two clusters with the same name in different zones would this work? I think the script should accept zone/region as an arg.

**Comment by rramkumar1** [comment]
When trying this locally I get the following error:

```
deployment.extensions/l7-default-backend configured
service/default-http-backend configured
/tmp/run-glbc.sh: line 24: --apiserver-host=http://localhost:7127: No such file or directory
Killing proxy (pid=125670)
```

File: hack/run-local-e2e.sh (hunk 2 above)
Comment (bowei) [reply hack/run-local-e2e.sh:1]: good point, fixing

File: hack/run-local-e2e.sh (hunk 1 above)
Comment (bowei) [reply hack/run-local-e2e.sh:23]: fixed

File: hack/run-local-e2e.sh (hunk 3 above)
Comment (bowei) [reply hack/run-local-e2e.sh:33]: added optional param

**Comment by bowei** [comment]
fixed issues, got rid of unneeded script generation

File: hack/run-local-glbc.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+GOOGLE_APPLICATION_CREDENTIALS="${HOME}/.config/gcloud/application_default_credentials.json"
+
+if [ ! -r ${GOOGLE_APPLICATION_CREDENTIALS} ]; then
+    echo "You must login your application default credentials"
+    echo "$ gcloud auth application-default login"
+    exit 1
+fi
+
+GCECONF=${GCECONF:-/tmp/gce.conf}
+GLBC=${GLBC:-./glbc}
```
Comment (rramkumar1) [inline hack/run-local-glbc.sh:16]: this seems to assume the glbc executable is in this directory?  If you run the script as is, it won't work unless you build first and move the executable here.

Comment (bowei) [reply hack/run-local-glbc.sh:16]: you can do that by

GLBC=mypath/xxx hack/run-local-glbc.sh

**Pull request merged**